pkgs/by-name/dw/dwl/package.nix

```diff
@@ -0,0 +1,125 @@
+{
+  lib,
+  fetchFromGitea,
+  installShellFiles,
+  libX11,
+  libinput,
+  libxcb,
+  libxkbcommon,
+  pixman,
+  pkg-config,
+  stdenv,
+  testers,
+  wayland,
+  wayland-protocols,
+  wayland-scanner,
+  wlroots,
+  writeText,
+  xcbutilwm,
+  xwayland,
+  # Boolean flags
+  enableXWayland ? true,
+  withCustomConfigDefH ? (configDefH != null || conf != null),
+  # Configurable options
+  configDefH ?
+    if conf != null then lib.warn "conf parameter is deprecated; use configDefH instead" conf else null,
+  conf ? null,
+}:
+
+# If we set withCustomConfigDefH, let's not forget configDefH
+assert withCustomConfigDefH -> (configDefH != null || conf != null);
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dwl";
+  version = "0.6";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "dwl";
+    repo = "dwl";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-fygUzEi4bgopesvHByfpatkLFYI98qozJOUBNM2t9Mg=";
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+    pkg-config
+    wayland-scanner
+  ];
+
+  buildInputs =
+    [
+      libinput
+      libxcb
+      libxkbcommon
+      pixman
+      wayland
+      wayland-protocols
+      wlroots
+    ]
+    ++ lib.optionals enableXWayland [
+      libX11
+      xcbutilwm
+      xwayland
+    ];
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  postPatch =
+    let
+      configFile =
+        if lib.isDerivation configDefH || builtins.isPath configDefH then
+          configDefH
+        else
+          writeText "config.def.h" configDefH;
+    in
+    lib.optionalString withCustomConfigDefH "cp ${configFile} config.def.h";
+
+  makeFlags =
+    [
+      "PKG_CONFIG=${stdenv.cc.targetPrefix}pkg-config"
+      "WAYLAND_SCANNER=wayland-scanner"
+      "PREFIX=$(out)"
+      "MANDIR=$(man)/share/man"
+    ]
+    ++ lib.optionals enableXWayland [
+      ''XWAYLAND="-DXWAYLAND"''
+      ''XLIBS="xcb xcb-icccm"''
+    ];
+
+  strictDeps = true;
+
+  # required for whitespaces in makeFlags
+  __structuredAttrs = true;
+
+  passthru = {
+    tests.version = testers.testVersion {
+      package = finalAttrs.finalPackage;
+      # dwl -v returns 1 and emits to stderr
+      command = "dwl -v 2>&1; return 0";
+    };
+  };
+
+  meta = {
+    homepage = "https://codeberg.org/dwl/dwl";
+    description = "Dynamic window manager for Wayland";
+    longDescription = ''
+      dwl is a compact, hackable compositor for Wayland based on wlroots. It is
+      intended to fill the same space in the Wayland world that dwm does in X11,
+      primarily in terms of philosophy, and secondarily in terms of
+      functionality. Like dwm, dwl is:
+
+      - Easy to understand, hack on, and extend with patches
+      - One C source file (or a very small number) configurable via config.h
+      - Limited to 2000 SLOC to promote hackability
```

Comment:
I am not sure the description is super important. But maybe it would be good to update it. This is not true anymore. From the readme:
> Given the Wayland architecture, dwl has to implement features from dwm and the xorg-server. Because of this, it is impossible to maintain the original project goal of 2000 SLOC and have a reasonably complete compositor with features comparable to dwm. However, this does not mean that the code will grow indiscriminately. We will try to keep the code as small as possible.

Reply:
Noted.
Fixed.